xchange-stream-btcmarkets/src/main/java/info/bitrich/xchangestream/btcmarkets/BTCMarketsStreamingService.java

```diff
@@ -5,66 +5,82 @@
 import com.google.common.collect.Sets;
 import info.bitrich.xchangestream.btcmarkets.dto.BTCMarketsWebSocketSubscribeMessage;
 import info.bitrich.xchangestream.service.netty.JsonNettyStreamingService;
+
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Set;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 class BTCMarketsStreamingService extends JsonNettyStreamingService {
-  static final String CHANNEL_ORDERBOOK = "orderbook";
-  static final String CHANNEL_HEARTBEAT = "heartbeat";
-  private static final Logger LOG = LoggerFactory.getLogger(BTCMarketsStreamingService.class);
-  private final Set<String> subscribedOrderbooks = Sets.newConcurrentHashSet();
+    static final String CHANNEL_ORDERBOOK = "orderbook";
+    static final String CHANNEL_TICK = "tick";
+    static final String CHANNEL_HEARTBEAT = "heartbeat";
+    private static final Logger LOG = LoggerFactory.getLogger(BTCMarketsStreamingService.class);
+    private final Set<String> subscribedOrderbooks = Sets.newConcurrentHashSet();
+    private final Set<String> subscribedtick = Sets.newConcurrentHashSet();
 
-  public BTCMarketsStreamingService(String apiUrl) {
-    super(apiUrl);
-  }
+    public BTCMarketsStreamingService(String apiUrl) {
+        super(apiUrl);
+    }
 
-  private BTCMarketsWebSocketSubscribeMessage buildSubscribeMessage() {
-    return new BTCMarketsWebSocketSubscribeMessage(
-        new ArrayList<>(subscribedOrderbooks),
-        Lists.newArrayList(CHANNEL_ORDERBOOK, CHANNEL_HEARTBEAT),
-        null,
-        null,
-        null);
-  }
+    private BTCMarketsWebSocketSubscribeMessage buildSubscribeMessage() {
+        return new BTCMarketsWebSocketSubscribeMessage(
```

Comment:
you still have not addressed this, you are using `subscribedtick`  for all of them, also please camelcase to follow java style

Reply:
Fixed

Reply:
This still is not quite right, if you think about this what is going to happen is if you subscribe for orderbook you will also end up subscribing to ticker. You need to compose this build subscribe message more conditionally.

You should test this locally if you are not already I would imagine you would see both ticker messages and orderbook messages and for someone subscribing on just one that is not the right behavior